lib/docs/core/doc.rb

```diff
@@ -48,7 +48,10 @@ def name
       end
 
       def slug
-        slug = @slug || name.try(:downcase)
+        slug = @slug || (
+          raise "Slug must be set explicitly when name (#{name}) consists of anything else than [\\w\\.%]" if /[^\w\.%]/ =~ name
+          name.try(:downcase)
+        )
```

Comment:
I think we can make this a bit simpler, like this:
```ruby
def slug
  slug = @slug || default_slug
  version? ? "#{slug}~#{version_slug}" : slug
end
```
then on L120 (after the `private`)
```ruby
def default_slug
  if name =~ /[^A-Za-z0-9_]/ 
    raise "slug must be set explicitly when name (#{name}) contains non-alphanumeric characters"
  end
  name.downcase
end
```